python/mxnet/io/io.py

```diff
@@ -748,11 +744,12 @@ def _batchify(self, data_source):
             self.cursor + self.batch_size > self.num_data:
             pad = self.batch_size - self.num_data + self.cursor
             first_data = self._getdata(data_source, start=self.cursor)
+            second_data = None
             if pad > self.num_data:
                 while True:
                     if pad <= self.num_data:
                         break
-                    second_data = self._getdata(data_source, end=self.num_data)
+                    second_data = self._concat(second_data, self._getdata(data_source, end=self.num_data))
```

Comment:
Can it be replaced with nd.repeat, in order to remove Python while-loop?

Reply:
Good idea